client/blocks/like-button/button.jsx

```diff
@@ -78,8 +78,22 @@ class LikeButton extends PureComponent {
 			containerClasses[ 'is-liked' ] = true;
 		}
 
+		const stopPropagation = ( event ) => {
+			if ( showLikeCount ) {
+				event.stopPropagation();
+			}
+		};
```

Comment:
Naming suggestion: this conditionally stops propagation, so it may be worth reflecting that in the function name.
```suggestion
		const maybeStopPropagation = ( event ) => {
			if ( showLikeCount ) {
				event.stopPropagation();
			}
		};
```